### ISSUE TITLE
[CockroachDB] Different behavior with CDB 21.1 than 20.2

### ISSUE THREAD

**Issue by r2liquibase** [liquibase/liquibase#1862]
## Environment
Windows

**Liquibase Version**:

4.3.5

**Liquibase Integration & Version**: <Pick one: CLI, maven, gradle, spring boot, servlet, etc.>

CLI

**Liquibase Extension(s) & Version**: 

N/A

**Database Vendor & Version**:

CockroachDB 21.1 and 20.2, both via Docker

**Operating System Type & Version**:

Windows 10

## Description
When running Liquibase against CDB 20.2, you get the following SQL from the changelog below:

`
CREATE TABLE public.TEST (id SERIAL NOT NULL, TEST VARCHAR(36) NOT NULL, TEST2 VARCHAR(50) NOT NULL, CONSTRAINT TEST_PKEY PRIMARY KEY (id), UNIQUE (TEST));
`

For 21.1, you get this:

`
CREATE TABLE public.TEST (id INTEGER GENERATED BY DEFAULT AS IDENTITY NOT NULL, TEST VARCHAR(36) NOT NULL, TEST2 VARCHAR(50) NOT NULL, CONSTRAINT TEST_PKEY PRIMARY KEY (id), UNIQUE (TEST));
`

It appears that Liquibase is sniffing for Database vendor and version number to direct behavior. Should be updated to to include 21.1, but I would argue it should encompass all versions of CDB, not specific ones. Pretty sure, they will always be backwards compatible with SQL just like all other DB vendors.

```
databaseChangeLog:
  - changeSet:
      id: 1
      author: test
      changes:
        - createTable:
            tableName: TEST
            columns:
              - column:
                  name: id
                  type: int
                  autoIncrement: true
                  constraints:
                    primaryKey: true
                    nullable: false
              - column:
                  name: TEST
                  type: varchar(36)
                  constraints:
                    nullable: false
                    unique: true
              - column:
                  name: TEST2
                  type: varchar(50)
                  constraints:
                    nullable: false
```

**Comment by r2liquibase** [comment]
I used the docker-compose.yml in liquibase-test-harness (commenting out all the other db's) and these .properties files

21.1
```
#liquibase.properties cdb 21.1
url: jdbc:postgresql://localhost:26259/lbcat
username: lbuser
password: 
changeLogFile: changelog.yml
```

20.2
```
#liquibase.properties cdb 20.2
url: jdbc:postgresql://localhost:26257/lbcat
username: lbuser
password: 	
changeLogFile: changelog.yml
```

**Comment by r2liquibase** [comment]
ok, i'll make it even easier:

docker-compose.yml

```
version: '3.3'

services:

  crdb-20.2:
    image: cockroachdb/cockroach:v20.2.0
    command: start-single-node --insecure
    expose:
      - "8080"
      - "26257"
    ports:
      - "26257:26257"
      - "8080:8080"
    networks:
      - roachnet
  crdb-20.2-init:
    image: cockroachdb/cockroach:v20.2.0
    networks:
      - roachnet
    volumes:
      - ./setup_db.sh:/setup_db.sh
    entrypoint: "/bin/bash"
    command: /setup_db.sh crdb-20.2
    depends_on:
      - crdb-20.2

  crdb-20.1:
    image: cockroachdb/cockroach:v20.1.8
    command: start-single-node --insecure
    expose:
      - "8080"
      - "26257"
    ports:
      - "26258:26257"
      - "8081:8080"
    networks:
      - roachnet
  crdb-20.1-init:
    image: cockroachdb/cockroach:v20.1.8
    networks:
      - roachnet
    volumes:
      - ./setup_db.sh:/setup_db.sh
    entrypoint: "/bin/bash"
    command: /setup_db.sh crdb-20.1
    depends_on:
      - crdb-20.1

  crdb-21.1:
    image: cockroachdb/cockroach:v21.1.0
    command: start-single-node --insecure
    expose:
      - "8080"
      - "26257"
    ports:
      - "26259:26257"
      - "8082:8080"
    networks:
      - roachnet
  crdb-21.1-init:
    image: cockroachdb/cockroach:v21.1.0
    networks:
      - roachnet
    volumes:
      - ./setup_db.sh:/setup_db.sh
    entrypoint: "/bin/bash"
    command: /setup_db.sh crdb-21.1
    depends_on:
      - crdb-21.1


networks:
  roachnet:

```

**Comment by r2liquibase** [comment]
Verified that `select version()` doesn't return weird stuff in CDB 21.1:

```
CockroachDB CCL v21.1.0 (x86_64-unknown-linux-gnu, built 2021/05/17 13:49:40, go1.15.11)
```

Here's 20.2:

```
CockroachDB CCL v20.1.8 (x86_64-unknown-linux-gnu, built 2020/10/21 15:46:38, go1.13.9)
```

I am mystified why Liquibase would not run the same SQL for CDB 21.1 as CDB 20.2

**Comment by kristinelson** [comment]
@r2liquibase any idea on timeline of when this will be fixed? I'm not able to update Cockroach to 21.1.x because of this known issue.

**Comment by kristinelson** [comment]
Fix raised here: https://github.com/liquibase/liquibase/pull/1908
